sim/sim.py

```diff
@@ -311,14 +332,29 @@ def power_in(self):
         return self.controller.power[self.myT]
     
      # ---
-    def data_rate(self, conditions = []):
+    def data_rate(self,conditions=[]):
```

Comment:
The right thing would be to add a time parameter here, like
```suggestion
    def data_rate(self,conditions=[], time_index):
```
and pass `myT` as the time_index